src/main/java/com/targus/ui/controllers/AlgorithmSelectionController.java

```diff
@@ -0,0 +1,58 @@
+package com.targus.ui.controllers;
+
+import com.targus.utils.Constants;
+import javafx.animation.KeyFrame;
+import javafx.animation.KeyValue;
+import javafx.animation.Timeline;
+import javafx.fxml.FXML;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+import javafx.util.Duration;
+
+public class AlgorithmSelectionController {
+    public ComboBox algorithmComboBox;
+    public VBox GAInputs;
+    public StackPane algorithmInputs;
+    public ComboBox mutationComboBox;
+    public TextField mutationRateTextField;
+    public ComboBox terminationComboBox;
+    public TextField terminationTextField;
+
+    public String getAlgorithm() {
+        return algorithmComboBox.getValue().toString();
+    }
+
+    public String getMutation() {
+        return mutationComboBox.getValue().toString();
+    }
+
+    public double getMutationRate() {
+        return Double.parseDouble(mutationRateTextField.getText());
+    }
+
+    public String getTermination() {
+        return terminationComboBox.getValue().toString();
+    }
+
+    public int getTerminationValue() {
+        return Integer.parseInt(terminationTextField.getText());
+    }
```

Comment:
Also, we need to make sure that these values are correct. No sophisticated validation process is necessary. If no value is provided, return 0 for example, and make sure the value is positive.

I think adding this simple validation process will not take too much of your time.

Reply:
Although I think what you did was a great remark, I have to disagree about the lack of need for a sophisticated validation process. Because for example if the user provides a negative value by the text field, the program will crash because it was not handled during the parsing process.